### PR TITLE
Inkline 3

### PR DESCRIPTION
**Pull request by alexgrozav**
Here you can track the progress on Inkline 3, the Vue 3 rewrite that will keep all of the awesome features of version 2, but has some huge improvements under the hood.

**Changelog:**
- Development workflow using Vite
- Build workflow using Vue CLI
- Hybrid SASS and CSS Variables with fine-grained design control
- Overall component UI, UX and DX improvements

### CODE REVIEW

File: packages/inkline3/.coveralls.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+service_name: travis-pro
```
Comment (TheJaredWilcurt) [inline packages/inkline3/.coveralls.yml:1]:
```suggestion
service_name: travis-pro

```

File: packages/inkline3/src/constants/keymap.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const keymap = {
+    tab: ['Tab', 9],
+    enter: ['Enter', 13],
+    esc: ['Escape', 27],
+    space: [' ', 'Space', 32],
+    left: ['ArrowLeft', 'Left', 37],
+    up: ['ArrowUp', 'Up', 38],
+    right: ['ArrowRight', 'Right', 39],
+    down: ['ArrowDown', 'Down', 40],
+};
```
Comment (TheJaredWilcurt) [inline packages/inkline3/src/constants/keymap.js:10]:
```suggestion
};

```

File: packages/inkline3/src/helpers/isVisible.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Check if given element is visible
+ *
+ * @param element
+ */
+export function isVisible(element) {
+    return Boolean(element) &&
+        Boolean(element.offsetWidth || element.offsetHeight || element.getClientRects().length);
+}
```
Comment (TheJaredWilcurt) [inline packages/inkline3/src/helpers/isVisible.js:9]:
```suggestion
}

```

File: packages/inkline3/src/helpers/once.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    const listener = function() {
+        if (fn) {
+            fn.apply(this, arguments);
+        }
+
+        off(el, event, listener);
+    };
+
+    on(el, event, listener);
+};
```
Comment (TheJaredWilcurt) [inline packages/inkline3/src/helpers/once.js:14]:
```suggestion
};

```

**Comment by calebeaires** [comment]
Is there some way to already use Inkline 3?

**Comment by alexgrozav** [comment]
I'm planning to release a beta version. However, the documentation would need to be run locally at this point. Would you like to try it out?

**Comment by calebeaires** [comment]
@alexgrozav - I do want to try.

**Comment by calebeaires** [comment]
I have tried to install it as a brand new project and it gets this error:

Install action 
`yarn add https://github.com/inkline/inkline.git#inkline3`


Error
```bash

error Can't add "root": invalid package version undefined.

```

**Comment by alexgrozav** [comment]
@calebeaires That's happening because it's nested inside a folder (due to the lerna monorepo structure).

I'll release `@inkline/inkline@3.0.0-beta.1` today. I'll write a message here. You'll be able to use it soon! :)

**Comment by calebeaires** [comment]
Testing v3. Thanks to the amazing framework!

**Comment by alexgrozav** [comment]
The pleasure is mine! :) I've updated the installation instructions for the beta repository. I'll let you know when the new docs are up.

Meanwhile, you can clone the repository and use `npm run dev` to launch the WIP docs.

**Pull request merged**